IdentityCore/src/oauth2/aad_base/MSIDAADTokenResponse.m

```diff
@@ -60,6 +61,11 @@ - (NSDate *)extendedExpiresOnDate
     return nil;
 }
 
+- (NSDate *)refreshOnDate
+{
+    if (self.refreshIn) return [NSDate dateWithTimeIntervalSinceNow:self.refreshIn];
```

Comment:
Should we check for the presence of refreshOn property here before we check for the refreshIn (e.g. check extendedExpiresOnDate implementation)?

Reply:
Are we expecting refresh_on in Token response. Current implementation assumes we are getting only refresh in value from server. refresh_On is set based on refresh in value while saving the access token to cache . Do we need to add refresh_on in response ?

Reply:
`Are we expecting refresh_on in Token response`

No.

`Do we need to add refresh_on in response ?`

The only concern is that you're calculating it dynamically every time. So every time you try to read the property, you'd get a different value.